openedx/api.py

```diff
@@ -368,9 +368,11 @@ def reconcile_edx_username(user, *, desired_username=None):
                     )
 
                 if not OpenEdxUser.objects.filter(
-                    edx_username=username_to_try
+                    edx_username=username_to_try,
+                    desired_edx_username=username_to_try,
```

Comment:
I don't think you need to filter on this because in your previous PR the unique constraint was removed so it's expected there might be duplicates here.